src/_pytest/outcomes.py

```diff
@@ -17,7 +18,9 @@ class OutcomeException(BaseException):
         contain info about test and collection outcomes.
     """
 
-    def __init__(self, msg: Optional[str] = None, pytrace: bool = True) -> None:
+    def __init__(
+        self, msg: Optional[Union[str, bytes]] = None, pytrace: bool = True
```

Comment:
i wonder if this one is actually a error in usage

Reply:
It seems to have been added in #1439 to fix #1178. Not sure if still relevant without Python 2 support.

I'll try to remove the `bytes` conversion in `__repr__` and run the tests.

Reply:
I added a commit which removes the `bytes` support. I'm mostly convinced it is useless now.